committer: rm useless extension conversion

jdodinh opened the pull request:
I will publish benchmark results shortly.

jdodinh commented:
Running the `whir_p3::whir::make_whir_things` test using Samply yields the following results on my machine (M3 Mac):
**Before the change:**
Total time to run `whir_p3::whir::make_whir_things`: 10.0 seconds (7139 samples)
52% spent in `whir_p3::whir::prover::Prover::prove` (3687 samples)
24% spent in `whir_p3::whir::verifier::Verifier::verify` (1693 samples)
11% spent in `whir_p3::whir::committer::writer::CommitmentWriter::commit` (790 samples)
<img width="658" alt="Screenshot 2025-04-30 at 16 10 24" src="https://github.com/user-attachments/assets/2ec70dd4-6bdd-44fc-8e5f-9fe5f072cadc" />

**After the change:**
Total time to run `whir_p3::whir::make_whir_things`: 7.9 seconds (7381 samples)
51% spent in `whir_p3::whir::prover::Prover::prove` (3753 samples)
22% spent in `whir_p3::whir::verifier::Verifier::verify` (1660 samples)
13% spent in `whir_p3::whir::committer::writer::CommitmentWriter::commit` (978 samples)
<img width="668" alt="Screenshot 2025-04-30 at 16 07 58" src="https://github.com/user-attachments/assets/97a095b6-d4f3-421b-959d-7a313309e086" />

jdodinh commented:
I'm attaching here Sample runs that you can upload to [https://profiler.firefox.com/](the profiler).

[Uploading after_1.json…]()

tcoratger commented on src/whir/prover.rs:
```diff
@@ -282,25 +293,57 @@ where
+                let commitment_merkle =
+                    MerkleTreeMmcs::new(self.0.merkle_hash.clone(), self.0.merkle_compress.clone());
+                for challenge in &stir_challenges_indexes {
+                    let (commitment_leaf, commitment_root): (Vec<Vec<F>>, Vec<[u8; DIGEST_ELEMS]>) =
```
```suggestion
                    let (commitment_leaf, commitment_root) =
```

tcoratger commented on src/whir/prover.rs:
```diff
@@ -282,25 +293,57 @@ where
+                let mut stir_evaluations = ood_answers;
+                let transformed: Vec<Vec<EF>> = answers
+                    .iter()
+                    .map(|inner| inner.into_iter().map(|&fel| EF::from(fel)).collect())
```
I think that we can keep that for a follow up but usually we should not due such a conversion because all the algebra is implemented for extension field with base field, so that it is usually better to keep things in the base field as much as we can.

tcoratger commented on src/whir/prover.rs:
```diff
@@ -282,25 +293,57 @@ where
+                stir_evaluations
+            }
+            Some(data) => {
+                let mut answers = Vec::new();
+                let mut merkle_proof = Vec::new();
```
Sam here can't we init with capacity?

tcoratger commented on src/whir/prover.rs:
```diff
@@ -392,15 +435,34 @@ where
+                let commitment_merkle =
+                    MerkleTreeMmcs::new(self.0.merkle_hash.clone(), self.0.merkle_compress.clone());
+                for challenge in final_challenge_indexes {
+                    let (commitment_leaf, commitment_root): (Vec<Vec<F>>, Vec<[u8; DIGEST_ELEMS]>) =
```
```suggestion
                    let (commitment_leaf, commitment_root) =
```

tcoratger commented on src/whir/prover.rs:
```diff
@@ -392,15 +435,34 @@ where
-        }
+        match &round_state.merkle_prover_data {
+            None => {
+                let mut answers = Vec::new();
+                let mut merkle_proof = Vec::new();
```
Can't we init with capacity?

tcoratger commented on src/whir/verifier.rs:
```diff
@@ -145,15 +149,42 @@ where
+                        .map_err(|_| ProofError::InvalidProof)?;
+                    stir_challenges_answers = answers
+                        .into_iter()
+                        .map(|inner| inner.into_iter().map(|&f_el| EF::from(f_el)).collect())
```
Same here we can keep for later as the scope of this PR is already sufficient but this conversion should be avoided for performance reasons.

tcoratger commented:
@jdodinh Thanks a lot for the PR, this is exactly what we want.
- I just did a first pass with a few minor comments.
- You can run the clippy locally to fix the small issues.

jdodinh commented:
Just addressed most of the comments. Regarding the conversions, I agree that we should avoid them as much as possible. The improvement is noticeable because we are only converting the necessary leaves, rather than the entire polynomial evaluation, but I think there is a way we can avoid it altogether. 

Also, a refactoring will be necessary, since currently there is a lot of repeated code

Would you like this to remain as part of this PR, or should we create a separate issue out of it?
